File: SwappingParentheses/solve/python/SwappingParenthesesSolver.py
#! /usr/bin/env python
import os, sys
import SwappingParentheses

def init():
    default_path = os.path.join(os.getenv("HOME"), ".gettc")
    gettc_home = os.path.abspath(os.getenv("GETTC_HOME", default_path))
    include_dir = os.path.join(gettc_home, "include/python")
    sys.path.append(include_dir)

def main():
    import topcoder as tc
    with open(sys.argv[1], "r") as fi:
        input = fi.read()
        reader = tc.Reader(input)

        s = reader.next("String")
        reader.next()
        op = reader.next("int[]")

    result = SwappingParentheses.countValid(s, op)
    with open(sys.argv[2], "w") as fo:
        fo.write(tc.write(result, "long"))

if __name__ == "__main__":
    init()
    main()

File: SwappingParentheses/solve/ruby/SwappingParenthesesSolver.rb
#! /usr/bin/env ruby

require_relative "SwappingParentheses"

require "gettc/types"
include Gettc

def init
  gettc_home = File.expand_path(ENV["GETTC_HOME"] || File.join(ENV["HOME"], ".gettc"))
  $LOAD_PATH << File.join(gettc_home, "include", "ruby")
  require "topcoder"
  include TopCoder
end

def main
  reader = Reader.new(IO.read(ARGV[0]))
  s = reader.next(TString)
  reader.next()
  op = reader.next(TArray.new(TInt))

  result = SwappingParentheses.new().countValid(s, op)
  IO.write(ARGV[1], Writer.new.next(result, TLong).to_s)
end

init
main
